### PR TITLE
Upgrade Chromium from 73.0.3683.75 to Chromium 74.0.3729.22

### PR DESCRIPTION
**Pull request by mkarolin**
Fix brave/brave-browser#3502
Related PR: brave/brave-browser#3790

It's easiest to review this per commit.
Each commit explains where the change came from in Chromium's history.

## Special notes to QA:

- Retest importer fully
- Retest ads fully
- Retest ledger fully
- Retest Tor fully
- Retest shields fully

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
See brave/brave-browser#3502.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: patches/third_party-blink-renderer-build-scripts-core-css-properties-templates-style_builder_functions.tmpl.patch [hunk 1]
```diff
@@ -1,31 +0,0 @@
-diff --git a/third_party/blink/renderer/build/scripts/core/css/properties/templates/style_builder_functions.tmpl b/third_party/blink/renderer/build/scripts/core/css/properties/templates/style_builder_functions.tmpl
-index 5ff29d8618a2261fcf92c9a04471d21ecef5b8af..17c2fb60f35793c30899505b7185d790dc026023 100644
---- a/third_party/blink/renderer/build/scripts/core/css/properties/templates/style_builder_functions.tmpl
-+++ b/third_party/blink/renderer/build/scripts/core/css/properties/templates/style_builder_functions.tmpl
-@@ -18,7 +18,7 @@ void {{prefix ~ '::' if prefix}}ApplyValue(StyleResolverState& state, const CSSV
-     {% if property.style_builder_inline == header and property.style_builder_generate_initial %}
- {{declare_initial(property.name.to_upper_camel_case() if not header)|indent(spaces, true)}} {
- {{caller(property)|indent(spaces, true)-}}
--{{'}'|indent(spaces, true)}}
-+{{'\n}'|indent(spaces, true)}}
```
Comment (mkarolin) [inline patches/third_party-blink-renderer-build-scripts-core-css-properties-templates-style_builder_functions.tmpl.patch]: The initial patch was:
<pre>
commit d1410d069d22188896cdc13a526b3732cc21c90b
Author: Brian R. Bondy
Date:   Fri Feb 15 00:03:08 2019 -0500

Add newline on line ending for } for generated file

This was causing an error on some computers when building (Anthony and me)
</pre>

In C74 the code has been changed to:
<pre>
    {% if property.style_builder_inline == header and property.style_builder_generate_initial %}
{{declare_initial(property.name.to_upper_camel_case() if not header)|indent(spaces, true)}} {
{{(caller(property) ~ '}')|indent(spaces, true)}}
</pre>

File: patches/chrome-browser-resources-history-history_toolbar.html.patch [hunk 2]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/resources/history/history_toolbar.html b/chrome/browser/resources/history/history_toolbar.html
```
Comment (mkarolin) [inline patches/chrome-browser-resources-history-history_toolbar.html.patch]: Moved from patches/chrome-browser-resources-md_history-history_toolbar.html.patch (deleted below).

File: patches/chrome-browser-net-net_error_diagnostics_dialog.h.patch [hunk 3]
```diff
@@ -1,19 +0,0 @@
...
- 
--// Returns true if the platform has a supported tool for diagnosing network
--// errors encountered when requesting URLs.
--bool CanShowNetworkDiagnosticsDialog();
-+// Returns true if a tool for diagnosing network errors encountered when
-+// requesting URLs can be shown for the provided WebContents. The ability to
-+// show the diagnostic tool depends on the host platform, and whether the
-+// WebContents is incognito.
-+bool CanShowNetworkDiagnosticsDialog(content::WebContents* web_contents);
-
```
Comment (bbondy) [inline patches/chrome-browser-net-net_error_diagnostics_dialog.h.patch]: cc @pilgrim-brave

Comment (bbondy) [reply patches/chrome-browser-net-net_error_diagnostics_dialog.h.patch]: Pls check here and other patches

File: chromium_src/net/socket/socks5_client_socket.cc [hunk 4]
```diff
@@ -10,7 +10,7 @@
 namespace net {
 
 int SOCKS5ClientSocket::DoAuth(int rv) {
-  rv = Authenticate(rv, *transport_, net_log_, io_callback_);
```
Comment (bbondy) [inline chromium_src/net/socket/socks5_client_socket.cc]: cc @yrliou pls review this commit: "Updated SOCKS5 client socket overrides."

File: browser/component_updater/brave_component_updater_configurator.cc [hunk 5]
```diff
@@ -22,10 +22,12 @@
 #include "components/prefs/pref_registry_simple.h"
```
Comment (bbondy) [inline browser/component_updater/brave_component_updater_configurator.cc]: cc @jumde please review this commit: "Change to classes inheriting from update_client::Configurator."

File: app/brave_main_delegate.cc [hunk 6]
```diff
@@ -145,6 +146,7 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
       autofill::features::kAutofillServerCommunication.name,
       features::kAudioServiceOutOfProcess.name,
       features::kDefaultEnableOopRasterization.name,
+      network::features::kNetworkService.name,
```
Comment (bbondy) [inline app/brave_main_delegate.cc]: I posted an issue here to revert this and support the network service:
https://github.com/brave/brave-browser/issues/3819

**Comment by bbondy** [comment]
I edited Comment 0 to add a section "Special notes to QA:" with some things I wanted them to pay attention to based on the code changes.

**Comment by bbondy** [comment]
Just noting that this will need an uplift to 0.63.x as well after it is merged to master. We should wait for Nightly builds for a couple days before merging in the Dev uplift though.

**Pull request merged**